pkg/controllers/management/auth/globalroles/fleetworkspace_binding_handler.go

```diff
@@ -0,0 +1,187 @@
+	globalRole, err := h.grCache.Get(globalRoleBinding.GlobalRoleName)
+	if err != nil {
+		return fmt.Errorf("unable to get globalRole: %w", err)
+	}
+	if globalRole.InheritedFleetWorkspacePermissions.WorkspaceVerbs == nil &&
+		globalRole.InheritedFleetWorkspacePermissions.ResourceRules == nil {
+		return nil
+	}
+	fleetWorkspaces, err := h.fwCache.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("unable to list fleetWorkspaces when reconciling globalRoleBinding %s: %w", globalRoleBinding.Name, err)
+	}
+
+	if err = h.reconcileResourceRulesBindings(globalRoleBinding, globalRole, fleetWorkspaces); err != nil {
+		return fmt.Errorf("error reconciling fleet permissions rules: %w", err)
+	}
+
+	if err = h.reconcileWorkspaceVerbsBindings(globalRoleBinding, globalRole, fleetWorkspaces); err != nil {
+		return fmt.Errorf("error reconciling fleet workspace verbs: %w", err)
+	}
+
+	return nil
+}
+
+func (h *fleetWorkspaceBindingHandler) reconcileResourceRulesBindings(globalRoleBinding *v3.GlobalRoleBinding, globalRole *v3.GlobalRole, fleetWorkspaces []*v3.FleetWorkspace) error {
+	var returnError error
+	subject := rbac.GetGRBSubject(globalRoleBinding)
+	roleref := v1.RoleRef{
+		APIGroup: v1.GroupName,
+		Kind:     "ClusterRole",
+		Name:     wrangler.SafeConcatName(globalRole.Name, fleetWorkspaceClusterRulesName),
+	}
+	subjects := []v1.Subject{
+		subject,
+	}
+
+	for _, fleetWorkspace := range fleetWorkspaces {
+		if fleetWorkspace.Name == localFleetWorkspace {
+			continue
+		}
+
+		rbName := wrangler.SafeConcatName(globalRoleBinding.Name)
+		rb, err := h.rbCache.Get(fleetWorkspace.Name, rbName)
+		if err != nil && !apierrors.IsNotFound(err) {
+			returnError = multierror.Append(returnError, err)
+			continue
+		}
+		if apierrors.IsNotFound(err) {
+			_, err = h.rbClient.Create(
+				&v1.RoleBinding{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      rbName,
+						Namespace: fleetWorkspace.Name,
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion: v3.GlobalRoleBindingGroupVersionKind.GroupVersion().String(),
+								Kind:       v3.GlobalRoleBindingGroupVersionKind.Kind,
+								Name:       globalRoleBinding.Name,
+								UID:        globalRoleBinding.UID,
+							},
+						},
+						Labels: map[string]string{
+							grbOwnerLabel:                 wrangler.SafeConcatName(globalRoleBinding.Name),
+							fleetWorkspacePermissionLabel: "true",
+							controllers.K8sManagedByKey:   controllers.ManagerValue,
+						},
+					},
+					RoleRef:  roleref,
+					Subjects: subjects,
+				})
+			if err != nil && !apierrors.IsNotFound(err) {
+				returnError = multierror.Append(returnError, err)
+			}
+		} else if !reflect.DeepEqual(rb.Subjects, subjects) ||
+			reflect.DeepEqual(rb.RoleRef, roleref) {
+			// undo modifications if rb has changed.
+			rb.Subjects = subjects
+			rb.RoleRef = roleref
```

Comment:
Since the roleRef field of a binding I think we don't need to check if the roleRef have changed. 

> Updating only when the subjects has changed, delete/create separately if the roleRef has changed
delete/create will need to be done by the user since roleRef is inmutable.

I would remove the roleRef, and just update if the subject has changed. What do you think?

Reply:
> > I would remove the roleRef, and just update if the subject has changed. What do you think?
> 
> Unfortunately I don't think that this will work. An end user could delete the original binding and create a new one with a different roleRef. This needs to correct such an adjustment.

ahh I see, but this is very unlikely to happen. If an end user deletes the original binding, the reconciliation loop will be triggered and the binding with the right roleRef would be created. The only scenario I see this can occur is if the controller is down and the user deletes and recreate the binding. 

I'll delete/create when any integrity check fails in order to cover this edge case.